Adding `functools.wraps` to `generate_calls()`.

dhermes opened the pull request:
Fixes #103.

----

For now, this has no tests, simply because I wanted to get it out for discussion with @theacodes. I am in the process of writing a few tests right now.

dhermes commented:
@theacodes I'm struggling here with the tests. I made an overly-specific update to `test__parametrize.py` but I think there needs to be an end-to-end test in place. Can you do the honor of deciding where it goes and how to run it? I'm happy to supply cases to be placed in there.

theacodes commented:
> Can you do the honor of deciding where it goes and how to run it?

Maybe makes sense for manifest to test the two combinations?

dhermes commented:
ISTM that `test_manifest.py` is too "unit testy", i.e. it invokes methods on `Manifest` directly, rather than running the whole system.

theacodes commented:
Eh, I'm fine with mixing "unit" tests and "integration" tests, as I really don't (personally) make a distinction.

dhermes commented:
@theacodes PTAL. I went a little wild and did some editorializing (that I maybe should put in a new PR):

- Added a LICENSE header to `tests/resources/noxfile.py`
- Removed your `nox._testing.Namespace` class and just replaced with `argparse.Namespace`.

Rather than putting the end-to-end test in `test_manifest.py` I put it in `test_main.py`. It (right now) has way too many magic strings in it, so we should discuss how to make the test less dependent on the contents of the source file. I **could** just dump contents in a tempfile, but I actually like the idea of having a sample repository of configs you could point users to.

theacodes commented on tests/resources/noxfile_nested.py:
```diff
@@ -12,8 +12,14 @@
-        self.__dict__.update(kwargs)
+
+def eat(cheese):
+    print("Noms, {} so good!".format(cheese))
```
if you're just printing, you don't need this indirection.

dhermes replied:
Yeah I saw that such calls were considered [legacy][1] now. I was just looking for an excuse to use the `session` object. Any better ideas?

[1]: https://github.com/theacodes/nox/blob/ad99797b630ca7abeb6bdf115ad0b4d788c30d26/nox/sessions.py#L158-L160

theacodes replied:
Eh, just leave it unused?

theacodes commented:
Looks fine, just one little comment. Definitely appreciate the cleanup work.

theacodes commented:
Go for it! Thank you! :)

On Thu, Aug 23, 2018 at 1:20 PM Danny Hermes <notifications@github.com>
wrote:

> *@dhermes* commented on this pull request.
> ------------------------------
>
> In tests/resources/noxfile_nested.py
> <https://github.com/theacodes/nox/pull/104#discussion_r212444011>:
>
> >
> -class Namespace:
> -    """A class that makes the provided kwargs available as attributes."""
> -    def __init__(self, **kwargs):
> -        self.__dict__.update(kwargs)
> +
> +def eat(cheese):
> +    print("Noms, {} so good!".format(cheese))
>
> Works for me. Dropped in 5c78529
> <https://github.com/theacodes/nox/commit/5c785299d74c77c9bc00ad8ee44cfb552cf1c523>.
> Shall I merge?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/theacodes/nox/pull/104#discussion_r212444011>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAPUc26yLsublkFAceKKiSJMdKR7lNS-ks5uTw6fgaJpZM4WIitr>
> .
>
